spec/ruby/core/enumerable/min_by_spec.rb

```diff
@@ -38,4 +37,36 @@
     multi = EnumerableSpecs::YieldsMulti.new
     multi.min_by {|e| e.size}.should == [1, 2]
   end
+
+ context "when called with an argument n" do
+   before :each do
+     @enum = EnumerableSpecs::Numerous.new(101, 55, 1, 20, 33, 500, 60)
+   end
+
+   context "without a block" do
+     it "returns an enumerator" do
+       @enum.min_by(2).should be_an_instance_of(enumerator_class)
+     end
+   end
+
+   context "with a block" do
+     it "returns an array containing the minimum n elements based on the block's value" do
+       result = @enum.min_by(3) { |i| i.to_s }
+       result.should == [1, 101, 20]
+     end
+
+     context "on a enumerable of length x where x < n" do
+       it "returns an array containing the minimum n elements of length n" do
+         result = @enum.min_by(500) { |i| i.to_s }
+         result.length.should == 7
+       end
+     end
+
+     context "when n is negative" do
+       it "raises an Argument error" do
+         lambda { @enum.min_by(-1) { |i| i.to_s } }.should raise_error(ArgumentError)
+       end
+     end
+   end
+ end
```

Comment:
In risk of sounding super pedantic, it looks like the indentation is a bit off here.